server/accounts.go

```diff
@@ -2936,17 +2941,33 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 			incompleteImports = append(incompleteImports, i)
 			continue
 		}
+		from := string(i.Subject)
+		to := i.GetTo()
 		switch i.Type {
 		case jwt.Stream:
-			s.Debugf("Adding stream import %s:%q for %s:%q", acc.Name, i.Subject, a.Name, i.To)
-			if err := a.AddStreamImportWithClaim(acc, string(i.Subject), string(i.To), i); err != nil {
-				s.Debugf("Error adding stream import to account [%s]: %v", a.Name, err.Error())
-				incompleteImports = append(incompleteImports, i)
+			if i.LocalSubject != "" {
+				// set local subject implies to is empty
+				to = string(i.LocalSubject)
+				s.Debugf("Adding stream import %s:%q for %s:%q", acc.Name, from, a.Name, to)
+				if err := a.AddMappedStreamImportWithClaim(acc, from, to, i); err != nil {
+					s.Debugf("Error adding stream import to account [%s]: %v", a.Name, err.Error())
+					incompleteImports = append(incompleteImports, i)
+				}
+			} else {
+				s.Debugf("Adding stream import %s:%q for %s:%q", acc.Name, from, a.Name, to)
+				if err := a.AddStreamImportWithClaim(acc, from, to, i); err != nil {
```

Comment:
Could have made the error handling (debug/adding to incompleteImeports slice) common to the if/else cases, but that's ok.

Reply:
Fixed